### ISSUE TITLE
Counting ReLU vs HardSwish FLOPs

### ISSUE THREAD

**Issue by jahongir7174** [huawei-noah/ghostnet#62]
Thank you very much for sharing the source code.
I have a question related to FLOPs counting for ReLU and HardSwish. I saw in the paper the flops are the same in ReLU and HardSwish.
Can you explain this situation?
![image](https://user-images.githubusercontent.com/71679734/111265500-8754be80-866c-11eb-8158-ad67f26b7af1.png)

**Comment by iamhankai** [comment]
I treat HSwish as zero FLOPs as MobileNetV3 does.

**Comment by jahongir7174** [comment]
I could not find any information about counting HardSwish as zero FLOPs in the paper and this might be my mistake. 
Can you share the point in the paper?

[this](https://github.com/rwightman/gen-efficientnet-pytorch/blob/master/BENCHMARK.md) file describes FLOPS of the models.
I found that FLOPS of mobilenetv3 has a similar value with paper

**Comment by iamhankai** [comment]
Sorry, I'm misleading. The FLOPS of HSwish is so small compared to that of Conv, its value is rounded-up.

**Comment by jahongir7174** [comment]
I calculated provided [ghostnet](https://github.com/huawei-noah/ghostnet/blob/master/ghostnet_pytorch/ghostnet.py) parameters and FLOPS below.
Number of parameters - 5171524
FLOP per operator type:
       0.279741 GFLOP.    98.7183%. Conv
       0.002561 GFLOP.   0.903756%. FC
    0.000563712 GFLOP.   0.198929%. Add
    0.000503328 GFLOP.    0.17762%. Mul
      3.936e-06 GFLOP. 0.00138898%. Div
              0 GFLOP.          0%. Concat
              0 GFLOP.          0%. Relu
       0.283373 GFLOP in Total

I replaced ReLU with HardSwish of provided ghostnet. I found that there is 0.007331 GFLOP difference.
[HardSwish](https://github.com/rwightman/pytorch-image-models/blob/14ac4abf74735102b50e78cc82a077fc12484da1/timm/models/layers/activations.py#L79)
FLOP per operator type:
       0.279741 GFLOP.    96.2288%. Conv
     0.00300736 GFLOP.    1.03451%. Add
     0.00294697 GFLOP.    1.01374%. Mul
       0.002561 GFLOP.   0.880965%. FC
     0.00244758 GFLOP.   0.841949%. Div
              0 GFLOP.          0%. Concat
       0.290704 GFLOP in Total

I counted FLOPS using [this](https://github.com/jahongir7174/FLOPS-pt)

**Comment by iamhankai** [comment]
I count FLOPS using https://github.com/Lyken17/pytorch-OpCounter and ignore BatchNorm since it can be fused into Conv during inference.

**Comment by jahongir7174** [comment]
Can you share a custom op for HardSwish?
[shared counter](https://github.com/Lyken17/pytorch-OpCounter) has no [operation counter](https://github.com/Lyken17/pytorch-OpCounter/blob/d1920d3400a62678ab3c2cf21ba13db35921713c/thop/profile.py#L25) for HardSwish.

**Comment by iamhankai** [comment]
HardSwish is formulated as `x = clip(x+3, 0, 6) / 6` where `+3` and `/6` can be fused into BN and Conv, so the FLOPS of HardSwish is the same as `clip`.

**Comment by jahongir7174** [comment]
Sorry for giving more questions, 
`x = clip(x+3, 0, 6) / 6` is [HardSwish](https://github.com/rwightman/pytorch-image-models/blob/14ac4abf74735102b50e78cc82a077fc12484da1/timm/models/layers/activations.py#L79) or [HardSigmoid](https://github.com/rwightman/pytorch-image-models/blob/14ac4abf74735102b50e78cc82a077fc12484da1/timm/models/layers/activations.py#L88)?

**Comment by iamhankai** [comment]
Sorry, It's HardSigmoid. HardSwish is `x*HardSigmoid(x)`. Implement its operation counter is easy.

**Comment by jahongir7174** [comment]
Thank you very much for your time
